### DISCUSSION TITLE
Is the shared store service write safe when multiple workers are writing to it using the same key ?

### DISCUSSION THREAD

**Discussion by SumanDas94** [webdriverio/webdriverio#12724] [Q&A]
I have a use case where multiple workers are generating data and writing to the shared store for the same key (it is a constraint that the key is the same for each worker). Is the service write safe in nature and will ensure that the write happening are atomic in nature ?

**Comment by christian-bromann** [comment]
Hey @SumanDas94 , we don't have any precautions in place to make the shared service write save while I haven't seen any issues being reported in that regard, I can't guarantee it. Contributions in that direction would be appreciated.
